pennylane/transforms/qcut.py

```diff
@@ -210,3 +213,235 @@ def tape_to_graph(tape: QuantumTape) -> MultiDiGraph:
             order += 1
 
     return graph
+
+
+@dataclass(frozen=True)
+class CutConfig:
+    """
+    Container wrapping partitioning parameters for passing to the automatic cutter.
+    """
+
+    k: int
+    imbalance: float
+    fragment_wires: List[int] = None
+    fragment_gates: List[int] = None
+
+
+def _infer_cut_configs(
+    num_tape_wires,
+    num_tape_gates,
+    max_device_wires,
+    max_device_gates,
+    min_device_wires=None,
+    min_device_gates=None,
+    k_lower=None,
+    k_upper=None,
+    imbalance_tolerance=None,
+    fragment_wires=None,
+    fragment_gates=None,
+):
+    """
+    Helper function for deriving the best default partitioning parameters for the automatic
+    circuit cutter. Arguments can be either user provided or inferred from devices.
+
+    Args:
+        num_tape_wires (int): Number of wires in the circuit tape to be partitioned.
+        num_tape_gates (int): Number of gates in the circuit tape to be partitioned.
+        max_device_wires (int): Number of wires for the largest available device.
+        max_device_gates (int): Maximum allowed circuit depth for the deepest available device.
+        min_device_wires (int): Number of wires for the smallest available device.
+            Optional, defaults to `max_device_wires`.
+        min_device_gates (int): Maximum allowed circuit depth for the shallowest available device.
+            Optional, defaults to `max_device_gates`.
+        k_lower (int): Lowest number of fragments to attempt for the automatic circuit cutter.
+            Optional, defaults to the scenario where each fragment is executed on the largest
+            available device, satisfying both wire and gate constraints.
+            If supplied, can be higher than said default if the the desired k is known.
+        k_upper (int): Largest number of fragments to attempt for the automatic circuit cutter.
+            Optional, defaults to the scenario where each fragment is executed on the smallest
+            available device, satisfying both wire and gate constraints.
+            If supplied, can be higher than said default to encourage exploration.
+        imbalance_tolerance (float): Maximum allowed imbalance for all partition attempts.
+            Can be used to globally override the imbalance constraint parameter.
+            Optional, defaults to a loose upper bound to avoid being too restrictive.
+        fragment_wires (List[int]): User provided predetermined list of fragment wire limits.
+            If supplied, k will be derived from it and exploration of other ks will not be made.
+        fragment_gates (List[int]): User provided predetermined list of fragment gate limits.
+            If supplied, k will be derived from it and exploration of other ks will not be made.
+
+    Returns:
+        cut_configs (List[CutConfig]): A list of cut configurations for passing to the automatic
+            circuit cutter to attempt.
```

Comment:
Since we are trying a range of k, and each kahypar call takes only a single k, we need a list of things we want to try. 

But we do have some freedom in how we do this. In its current state, each `CutConfig` corresponds to a single call to kahypar with fixed k and precalculated imbalance (depending on k), etc. An alternative is to generate them dynamically, which will require moving all the arguments and some logic of the helper function into the class. But let's not over-engineer this 😆